docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -58,72 +62,72 @@ trait ERC20ABI {
 :erc20-supply: xref:/guides/erc20-supply.adoc[Creating ERC20 Supply]
 
-WARNING: The following example uses `unsafe_new_contract_state` to access another contract's state.
-Although this is useful to use them as modules, it's considered unsafe because storage members could clash among used contracts if not reviewed carefully.
-Extensibility will be revisited after {components} are introduced.
-
-Using Contracts for Cairo, constructing an ERC20 contract requires setting up the constructor and exposing the ERC20 interface.
+Using Contracts for Cairo, constructing an ERC20 contract requires setting up the constructor and instantiating the ERC20 implementation.
 Here's what that looks like:
 
 [,javascript]
 ----
 #[starknet::contract]
 mod MyToken {
+    use openzeppelin::token::erc20::ERC20 as erc20_component;
+    use openzeppelin::token::erc20::interface;
     use starknet::ContractAddress;
-    use openzeppelin::token::erc20::ERC20;
+
+    component!(path: erc20_component, storage: erc20, event: ERC20Event);
+
+    #[abi(embed_v0)]
+    impl ERC20Impl = erc20_component::ERC20Impl<ContractState>;
+    #[abi(embed_v0)]
+    impl ERC20MetadataImpl = erc20_component::ERC20MetadataImpl<ContractState>;
+    impl ERC20InternalImpl = erc20_component::InternalImpl<ContractState>;
 
     #[storage]
-    struct Storage {}
+    struct Storage {
+        #[substorage(v0)]
+        erc20: erc20_component::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        ERC20Event: erc20_comp::Event
+    }
 
     #[constructor]
     fn constructor(
-        self: @ContractState,
+        ref self: ContractState,
         initial_supply: u256,
         recipient: ContractAddress
     ) {
         let name = 'MyToken';
         let symbol = 'MTK';
 
-        let mut unsafe_state = ERC20::unsafe_new_contract_state();
-        ERC20::InternalImpl::initializer(ref unsafe_state, name, symbol);
-        ERC20::InternalImpl::_mint(ref unsafe_state, recipient, initial_supply);
-    }
-
-    #[external(v0)]
-    impl MyTokenImpl of IERC20<ContractState> {
-        fn name(self: @ContractState) -> felt252 {
-            let mut unsafe_state = ERC20::unsafe_new_contract_state();
-            ERC20::ERC20Impl::name(@unsafe_state)
-        }
-
-        (...)
+        self.erc20.initializer(name, symbol);
+        self.erc20._mint(recipient, initial_supply);
     }
 }
 ----
 
-In order for the `MyToken` contract to extend the ERC20 contract, it utilizes the `unsafe_new_contract_state`.
-The unsafe contract state allows access to ERC20's implementations.
-With this access, the constructor first calls the initializer to set the token name and symbol.
-The constructor then calls `_mint` to create a fixed supply of tokens.
+In order for the `MyToken` contract to integrate the ERC20 component, it instantiates both the `ERC20Impl` and `ERC20MetadataImpl`.
```

Comment:
Agreed, it is inconsistent. I edited the doc so that the general ERC20 references don't include back ticks but the specific `ERC20` refs do. Let me know what you think

Reply:
For general, I mean not referencing anything specific to Contracts for Cairo e.g. "To support older ERC20 token deployments..."

And for specific, I mean referencing the Contracts for Cairo ERC20 mod/impl/interface e.g. "The `ERC20` component includes..."

Reply:
to me the backticks only represents code entities, and without i'd interpret it as the ERC20 standard. I think this is close to what you mean, but I don't think readers will catch the OZ/non-OZ distinction.

Reply:
> but I don't think readers will catch the OZ/non-OZ distinction

Fair enough, I'll edit and improve the content